routing/pathfind_test.go

```diff
@@ -3502,11 +3502,11 @@ func TestLastHopPayloadSize(t *testing.T) {
 			t.Parallel()
 
 			var mpp *record.MPP
-			if tc.restrictions.PaymentAddr != nil {
-				mpp = record.NewMPP(
-					tc.amount, *tc.restrictions.PaymentAddr,
-				)
-			}
+			tc.restrictions.PaymentAddr.WhenSome(
```

Comment:
I think could also be: 
```go
mpp := fn.MapOptionZ(func(p [32]byte) *record.MPP {
    return record.NewMPP(tc.amount, addr)
})
```

Reply:
oooo cool - TIL

Reply:
Yeah the diff with this one is that if the option is `None`, then it'll return the zero value of the return type. So in this case, it'll just return `nil` if the option isn't set.